core/commands/inline_diff.py

```diff
@@ -219,26 +219,33 @@ def get_object_contents(self, object_hash):
         """
         return self.git("show", "--no-color", object_hash)
 
+    def get_file_contents_binary(self, file_path):
+        """
+        Given an absolute file path, return the binary contents of that file
+        as a string.
+        """
+        file_path = os.path.join(self.repo_path, file_path)
+        with open(file_path, "rb") as f:
+            return f.read()
```

Comment:
Inline diff shows the whole file as changed because of different line feeds. On Windows the `core.autocrlf` setting is set `true` which causes files in the working tree to have Windows style `\r\n` line endings while the file in the index still has `\n`. By passing the binary content to `hash-object` the translation is no longer done and the file is "commited" as is. I changed the following line just to see if this is the case ... and it is. With those lines, everything is fine for me. Not sure, if there any edge cases to care about, when replacing.

```python
            binary = f.read()
            binary = binary.replace(b"\r\n", b"\n")
            binary = binary.replace(b"\r", b"")
            return binary
```